gradle-jdks-excavator-configurations/src/main/java/com/palantir/gradle/jdks/GradleJdksConfigurator.java

```diff
@@ -0,0 +1,100 @@
+     * @param caCerts the list of certificates that need to be added to the jdk. This influences the hash and the
+     * local installation directory name.
+     */
+    public static void writeGradleJdkInstallationConfigs(
+            Path targetDir,
+            String baseUrl,
+            JdkDistributionName jdkDistributionName,
+            Os os,
+            Arch arch,
+            String javaVersion,
+            String jdkVersion,
+            Map<String, String> caCerts) {
+        JdkSpec jdkSpec = JdkSpec.builder()
+                .distributionName(jdkDistributionName)
+                .release(JdkRelease.builder()
+                        .arch(arch)
+                        .os(os)
+                        .version(jdkVersion)
+                        .build())
+                .caCerts(CaCerts.from(caCerts))
+                .build();
+
+        Path jdksDir = targetDir.resolve("jdks");
+        Path jdkOsArchDir = jdksDir.resolve(javaVersion).resolve(os.uiName()).resolve(arch.uiName());
+        try {
+            Files.createDirectories(jdkOsArchDir);
+            GradleJdksConfigsUtils.writeConfigurationFile(
+                    jdkOsArchDir.resolve("download-url"), resolveDownloadUrl(baseUrl, jdkSpec));
+            GradleJdksConfigsUtils.writeConfigurationFile(
+                    jdkOsArchDir.resolve("local-path"), resolveLocalPath(jdkSpec));
+        } catch (Exception e) {
+            throw new RuntimeException(
+                    String.format("Failed to crate jdk configuration files in dir %s", jdkOsArchDir), e);
+        }
+    }
+
+    private static String resolveLocalPath(JdkSpec jdkSpec) {
+        return String.format(
+                "%s-%s-%s", jdkSpec.distributionName(), jdkSpec.release().version(), jdkSpec.consistentShortHash());
+    }
+
+    private static String resolveDownloadUrl(String baseUrl, JdkSpec jdkSpec) {
+        JdkPath jdkPath = jdkDistributions.get(jdkSpec.distributionName()).path(jdkSpec.release());
+        return String.format("%s/%s.%s", baseUrl, jdkPath.filename(), jdkPath.extension());
+    }
+
+    /**
+     * Writes the installation scripts & the jars to the target directory.
+     * @param targetDir the target directory where the installation scripts should be written to.
+     */
+    public static void writeInstallationScripts(Path targetDir) {
+        Path scriptsDir = targetDir.resolve("scripts");
+        GradleJdksConfigsUtils.createDirectories(scriptsDir);
+        Path functionsScript = GradleJdksConfigsUtils.copyResourceToPath(scriptsDir, "gradle-jdks-functions.sh");
+        GradleJdksConfigsUtils.setExecuteFilePermissions(functionsScript);
+        Path installationScript = GradleJdksConfigsUtils.copyResourceToPath(scriptsDir, "install-jdks.sh");
+        GradleJdksConfigsUtils.setExecuteFilePermissions(installationScript);
+        GradleJdksConfigsUtils.copyResourceToPath(scriptsDir, "gradle-jdks-setup.jar");
```

Comment:
This is interesting - I assumed we were going to excavate out a version number to each repo and have them curl some tar, extract it then run the script. I think your change may be nicer though as we just extract everything the repo needs into the repo then they run it.

I think the interesting part is how we ensure we always update these files to latest. We can add this as a library to excavator but we need to make sure it keeps up to date. Otherwise we could download the latest version of this jar and load it in a classloader or something...

Reply:
Actually, I'm being silly excavator doesn't need releases and elevator will ensure it upgrades fine. Using as a library will work just fine.